src/frontend/src/components/FilterPanel/components/DateFilter/index.tsx

```diff
@@ -79,22 +86,31 @@ const DateFilter: FC<Props> = ({
     return d.toISOString().substring(0, 10);
   };
 
+  // Use this over directly using `updateDateFilter` prop so we track filter changes.
```

Comment:
jw, did you also test to make sure this change did not impact other places where this input is used (ie: upload page 2 for meta data)?